src/Compilers/CSharp/Portable/Syntax/ClassDeclarationSyntax.cs

```diff
@@ -0,0 +1,141 @@
+        public static ClassDeclarationSyntax ClassDeclaration(
+            SyntaxList<AttributeListSyntax> attributeLists,
+            SyntaxTokenList modifiers,
+            SyntaxToken keyword,
+            SyntaxToken identifier,
+            TypeParameterListSyntax? typeParameterList,
+            BaseListSyntax? baseList,
+            SyntaxList<TypeParameterConstraintClauseSyntax> constraintClauses,
+            SyntaxToken openBraceToken,
+            SyntaxList<MemberDeclarationSyntax> members,
+            SyntaxToken closeBraceToken,
+            SyntaxToken semicolonToken)
+            => SyntaxFactory.ClassDeclaration(
+                attributeLists,
+                modifiers,
+                keyword,
+                identifier,
+                typeParameterList,
+                parameterList: null,
+                baseList,
+                constraintClauses,
+                openBraceToken,
+                members,
+                closeBraceToken,
+                semicolonToken);
+
+        /// <summary>Creates a new ClassDeclarationSyntax instance.</summary>
+        public static ClassDeclarationSyntax ClassDeclaration(
+            SyntaxList<AttributeListSyntax> attributeLists,
+            SyntaxTokenList modifiers,
+            SyntaxToken identifier,
+            TypeParameterListSyntax typeParameterList,
+            BaseListSyntax baseList,
+            SyntaxList<TypeParameterConstraintClauseSyntax> constraintClauses,
+            SyntaxList<MemberDeclarationSyntax> members)
+            => SyntaxFactory.ClassDeclaration(
+                attributeLists,
+                modifiers,
+                SyntaxFactory.Token(SyntaxKind.ClassKeyword),
+                identifier,
+                typeParameterList,
+                baseList,
+                constraintClauses,
+                SyntaxFactory.Token(SyntaxKind.OpenBraceToken),
+                members,
+                SyntaxFactory.Token(SyntaxKind.CloseBraceToken),
+                semicolonToken: default);
+
+        /// <summary>Creates a new ClassDeclarationSyntax instance.</summary>
+        public static ClassDeclarationSyntax ClassDeclaration(
+            SyntaxList<AttributeListSyntax> attributeLists,
+            SyntaxTokenList modifiers,
+            SyntaxToken identifier,
+            TypeParameterListSyntax? typeParameterList,
+            ParameterListSyntax? parameterList,
+            BaseListSyntax? baseList,
+            SyntaxList<TypeParameterConstraintClauseSyntax> constraintClauses,
+            SyntaxList<MemberDeclarationSyntax> members)
+            => SyntaxFactory.ClassDeclaration(
+                attributeLists,
+                modifiers,
+                SyntaxFactory.Token(SyntaxKind.ClassKeyword),
+                identifier,
+                typeParameterList,
+                parameterList,
+                baseList,
+                constraintClauses,
+                openBraceToken: default,
```

Comment:
>default [](start = 32, length = 7)

This doesn't look right.  It creates a node without braces nor semicolon.

Reply:
However, since you don't know the language version at this point, perhaps this should create brace tokens always.

---
In reply to: [361762456](https://github.com/dotnet/roslyn/pull/40467#discussion_r361762456) [](ancestors = 361762456)